tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -985,8 +1057,11 @@ class SerializableModelLibraryGenerator {
     List<SerializableModelFieldDefinition> fields,
     String? tableName, {
     required bool setAsToThis,
+    List<SerializableModelFieldDefinition>? parentClassFields,
   }) {
-    return fields
+    var inheritedFields = parentClassFields ?? [];
```

Comment:
Suggestion: maybe this should also be part of the classDefinition? It seems like we are repeating this logic in a lot of places in the library generator as well?

Maybe these should simply be part of the "fields" for the class? Or is it valuable to differentiate between the parent fields and the class fields?

Reply:
Yeah this is something we can surely improve by moving this to `ClassDefinition`. :muscle: 

We kind of need to differentiate between the fields of the class and the `parentFields` as class-fields are passed to `this` and parentFields are passed to `super` in the constructor